### DISCUSSION TITLE
Surface Book 2 Freezes When Detaching

### DISCUSSION THREAD

**Discussion by xharr1** [linux-surface/linux-surface#590] [Q&A] [answered]
First of all, thanks for this, it has been an awesome experience overall. 

As stated in the title, my surface book 2 is freezing when detaching from the base. It stays frozen after reattaching as well. I followed the Ubuntu/Debian portion of the installation guide and I have confirmed that the output of `uname -a` contains the string "surface" as specified in the installation guide: <computer name> 5.14.8-surface #1 SMP Mon Sep 27 15:43:15 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux.

I have also installed the surface-dtx-daemon, and left the config settings for that at the defaults.

Here is confirmation of my Ubuntu version as well:
Distributor ID:	Ubuntu
Description:	Ubuntu 20.04.3 LTS
Release:	20.04

Let me know if you have any ideas that I could test to get this working. Thanks again!

**Comment by qzed** [comment, marked as answer]
I assume the nvidia or nouveau driver is active. That can also happen if you're not actively using the dGPU, for example it can also be caused by the desktop environment (aka DE; e.g. Gnome) loading when the dGPU is not explicitly disabled. If you detach when the dGPU is in use (either direct or indirect), the DE can crash, leading to what you're experiencing.

So, in essence, to successfully detach you need to unload the drivers before. This might not always work though since the DE might not allow that (Gnome has been a bit picky about that in the past). Alternatively, you can disable the dGPU and only enable it when you actually need it (see e.g. https://github.com/linux-surface/surface-hotplug/wiki/Runtime-Power-Management#configuring-the-nvidia-driver).
